core/commands/partials/vagrant_terraform_configuration_generator.rb

```diff
@@ -1,22 +1,17 @@
 # frozen_string_literal: true
 
-require 'date'
 require 'fileutils'
 require 'json'
-require 'pathname'
-require 'securerandom'
-require 'socket'
-require 'erb'
 require 'set'
 require_relative '../base_command'
-require_relative '../../out'
 require_relative '../../models/configuration.rb'
-require_relative '../../services/shell_commands'
 require_relative '../../../core/services/configuration_generator'
+require_relative 'vagrantfile_generator'
+require_relative 'aws_terraform_generator'
+require_relative '../../../core/services/machine_commands'
 
-# The class generates the MDBCI configuration for use in pair with the Vagrant backend
-# rubocop:disable Metrics/ClassLength
-class VagrantConfigurationGenerator < BaseCommand
+# The class generates the MDBCI configuration for use in pair with the Vagrant or Terraform backend
+class VagrantTerraformConfigurationGenerator < BaseCommand
```

Comment:
There must be two separate classes VagrantConfigurationGenerator and TerraformConfigurationGenerator